crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -102,6 +101,58 @@ impl Consensus {
         sha256::Hash::from_slice(leaf_hash.as_slice())
             .expect("parent_hash: Engines shouldn't be Err")
     }
+
+    /// Check whether a block is valid
+    pub fn validate_block(
+        &self,
+        block: &bitcoin::Block,
+        ancestor: &BlockHeader,
+        height: u32,
+        inputs: HashMap<bitcoin::OutPoint, bitcoin::TxOut>,
+        flags: c_uint,
+    ) -> Result<(), BlockchainError> {
```

Comment:
`validate_block()` contains all necessary validation. Every method trying to validate any block should call this.